src/api/app/views/webui2/webui/kiwi/_tabs.html.haml

```diff
@@ -0,0 +1,37 @@
+= content_for :ready_function do
+  $("#kiwi_details_trigger").click(function() {
```

Comment:
Please use dashes ('-') for html classes and ids

Reply:
I wonder why HAML [IdName cop]https://github.com/brigade/haml-lint/blob/master/lib/haml_lint/linter/README.md#idnames) is not complaining about this :thinking:

Reply:
I found out! :nerd_face: It because of the way this file is structured. I didn't realised when reviewing it the first time but it should be:

```haml
- @project = @image.package.try(:project)
- @package = @image.package

.bg-light
  %ul.nav.nav-tabs.pt-2.px-3.flex-nowrap.collapsible{ 'role': 'tablist' }
    %li.nav-item
      = link_to('Overview', { controller: 'kiwi/images', action: 'show' }, class: "nav-link #{'active' if action_name == 'show'}")
    %li.nav-item
      = link_to 'Details', action_name == 'edit' ? '#' : { controller: 'kiwi/images', action: 'edit', section: 'details' },
        id: 'kiwi_details_trigger', class: "nav-link #{'active' if params[:section] == 'details'}"
    %li.nav-item
      = link_to 'Software', action_name == 'edit' ? '#' : { controller: 'kiwi/images', action: 'edit', section: 'software' },
        id: 'kiwi_software_trigger', class: "nav-link #{'active' if params[:section] == 'software'}"

= content_for :ready_function do
  $("#kiwi_details_trigger").click(function() {
  $("#kiwi-image-details-section").removeClass('d-none');
  $("#kiwi-preferences").removeClass('d-none');
  $(".detailed-info").removeClass('d-none');
  $("#kiwi-image-profiles-section").addClass('d-none');
  $("#kiwi-image-repositories-section").addClass('d-none');
  $("#kiwi-image-packages-section").addClass('d-none');
  $("#kiwi_software_trigger").removeClass("active");
  $("#kiwi_details_trigger").addClass("active");
  });

  $("#kiwi_software_trigger").click(function() {
  $("#kiwi-image-profiles-section").removeClass('d-none');
  $("#kiwi-image-repositories-section").removeClass('d-none');
  $("#kiwi-image-packages-section").removeClass('d-none');
  $("#kiwi-preferences").addClass('d-none');
  $(".detailed-info").addClass('d-none');
  $("#kiwi-image-details-section").addClass('d-none');
  $("#kiwi_software_trigger").addClass("active");
  $("#kiwi_details_trigger").removeClass("active");
  });
```

So you don't need the `:ruby` section, but to add an `-` before the line so that is executed but not printed. After that the liner will understand the code and start complaining! :bowtie: We normally put the javascript at the end, I think it is easier to read, as the ids and classes you use in the javascript are defined in the javascript and consecuently they should be written first. :wink:

Reply:
well, I was not completely right. The offense @bgeuken is complaining about, is actually in the Ruby code, so haml-lint can not complain about. It also not a Rubocop offense, as there is not a cop in Rubocop for this. So basically, no linter is responsible for it.

But there are other offenses which do appear if you write the code as I say. I think this is a bug in haml-lint: https://github.com/brigade/haml-lint/issues/287 But still I think we shouldn't use `:ruby` because of consistency and readability (I hope we can enforce this soon: https://github.com/brigade/haml-lint/issues/288)